### ISSUE TITLE
HEVC encoder can generate incorrect bitstream sometimes

### ISSUE THREAD

**Issue by chimojs** [GPUOpen-LibrariesAndSDKs/AMF#257]
I'm using SimpleEncoder demo in version *1.4.14* and find that if I change codec  parameter to **HEVC, 1280x720, 20fps**, the resulted h265 bitstream can not be played by ffplay.  After some debugging, I find ffmpeg can not parse this bitstream because there is a sequence **0x00 0x00 0x02** in sps parameter set without escaping.  Is it fine to have bitstream like this ?  
![bitstream](https://user-images.githubusercontent.com/25942744/115483651-c5ed2280-a283-11eb-93f6-8d673e0b6f61.png)

**Comment by chimojs** [comment]
Platform: Win10
Hardware: AMD Radeon RX 580 2048SP
Driver Version: Radeon software 20.12.1

**Comment by MikhailAMD** [comment]
You can share the stream - I will check it with analyzer.  You can also share your modified sample code. 
In my test I also see 00 00 02 value but Vega analyzer confirms validity of the stream.

**Comment by chimojs** [comment]
This is the generated bitstream [output.zip](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/files/6354769/output.zip).  Git diff output of modified code is here:
```
-static const wchar_t *pCodec = AMFVideoEncoderVCE_AVC;
-//static const wchar_t *pCodec = AMFVideoEncoder_HEVC;
+//static const wchar_t *pCodec = AMFVideoEncoderVCE_AVC;
+static const wchar_t *pCodec = AMFVideoEncoder_HEVC;
 
  //#define ENABLE_4K
 
 #ifdef _WIN32
-static amf::AMF_MEMORY_TYPE memoryTypeIn  = amf::AMF_MEMORY_DX9;
+static amf::AMF_MEMORY_TYPE memoryTypeIn  = amf::AMF_MEMORY_DX11;
 #else
 static amf::AMF_MEMORY_TYPE memoryTypeIn  = amf::AMF_MEMORY_VULKAN;
 #endif
@@ -65,18 +65,18 @@ static amf::AMF_SURFACE_FORMAT formatIn   = amf::AMF_SURFACE_NV12;
 static amf_int32 widthIn                  = 1920*2;
 static amf_int32 heightIn                 = 1080*2;
 #else
-static amf_int32 widthIn                  = 1920;
-static amf_int32 heightIn                 = 1080;
+static amf_int32 widthIn                  = 1280;
+static amf_int32 heightIn                 = 720;
 #endif
-static amf_int32 frameRateIn              = 30;
+static amf_int32 frameRateIn              = 20;
 static amf_int64 bitRateIn                = 5000000L; // in bits, 5MBit
 static amf_int32 rectSize                 = 50;
-static amf_int32 frameCount               = 500;
+static amf_int32 frameCount               = 5;
 static bool bMaximumSpeed = true;
 
 #define START_TIME_PROPERTY L"StartTimeProperty" // custom property ID to store submission time in a frame - all custom properties are copied from input to output
 
-static const wchar_t *fileNameOut = L"./output.h264";
+static const wchar_t *fileNameOut = L"./output.h265";
```
This is the whole `SimpleEncoder.cpp`   [SimpleEncoder.zip](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/files/6354780/SimpleEncoder.zip)

**Comment by MikhailAMD** [comment]
While Analyzer doesn't complain, looks like a problem, I've opened an internal ticket. Thanks.

**Comment by MikhailAMD** [comment]
This should be fixed with the latest driver release (21.6.1)

**Comment by chimojs** [comment]
Hi, 
I updated my driver to this version today and got the exactly same bitstream as before. Am I missing something?

**Comment by MikhailAMD** [comment]
Hi, confirmed that the fix missed this release. I am pushing the change to a hotfix release. Will update when released, shouldn't be long. Sorry about this.

**Comment by MikhailAMD** [comment]
The fix should be in the latest posted driver 21.7.1.

**Comment by chimojs** [comment]
Yes, I can play this bitstream without any problem with ffplay. Thanks, Mikhail.
